pkger/parser.go

```diff
@@ -102,8 +104,71 @@ func Parse(encoding Encoding, readerFn ReaderFn, opts ...ValidateOptFn) (*Templa
 	return pkg, nil
 }
 
-// FromFile reads a file from disk and provides a reader from it.
-func FromFile(filePath string) ReaderFn {
+// limitReadFileMaxSize is the maximum file size that limitReadFile will read.
+const limitReadFileMaxSize int64 = 2 * 1024 * 1024
+
+// limitReadFile operates like ioutil.ReadFile() in that it reads the contents
+// of a file into RAM, but will only read regular files up to the specified
+// max. limitReadFile reads the file named by filename and returns the
+// contents. A successful call returns err == nil, not err == EOF. Because
+// limitReadFile reads the whole file, it does not treat an EOF from Read as an
+// error to be reported.
+//
+// Ultimately, only remocal will have file:// URLs enabled for its 'data
+// catalogs' bootstrap functionality. At present, the largest catalog is 40k,
+// so set a small max here with a bit of headroom.
+func limitReadFile(name string) (buf []byte, rErr error) {
+	// use os.Open() to avoid TOCTOU
+	f, err := os.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer caperr.Capture(&rErr, f.Close)
+
+	// Check that properties of file are OK.
+	st, err := f.Stat()
+	if err != nil {
+		return nil, err
+	}
+
+	// Disallow reading from special file systems (e.g. /proc, /sys/, /dev).
+	if special, err := fs.IsSpecialFSFromFileInfo(st); err != nil {
+		return nil, fmt.Errorf("%w: %q", err, name)
+	} else if special {
+		return nil, fmt.Errorf("file in special file system: %q", name)
+	}
+
+	// only support reading regular files
+	if st.Mode()&os.ModeType != 0 {
+		return nil, fmt.Errorf("not a regular file: %q", name)
+	}
+
+	// limit how much we read into RAM
+	var size int
+	size64 := st.Size()
+	if size64 > limitReadFileMaxSize {
```

Comment:
Do we want to add a `&& limitReadFileMaxSize > 0` in case it becomes a variable some day?  Not blocking on merge, just a thought.

Reply:
Added in latest commit.